core/src/types/response.rs

```diff
@@ -62,6 +62,30 @@ impl Output {
 			error: Error::new(ErrorCode::InvalidRequest),
 		})
 	}
+
+	/// Get the jsonrpc protocol version.
+	pub fn version(&self) -> Option<Version> {
+		match *self {
+			Output::Success(ref s) => s.jsonrpc,
+			Output::Failure(ref f) => f.jsonrpc,
+		}
+	}
+
+	/// Get the correlation id.
+	pub fn id(&self) -> &Id {
+		match *self {
+			Output::Success(ref s) => &s.id,
+			Output::Failure(ref f) => &f.id,
+		}
+	}
+
+	/// Convert into a result. Will be `Ok` if it is a `Success` and `Err` if `Failure`.
+	pub fn into_result(self) -> Result<Value, Error> {
```

Comment:
Why not `Into<Result<V, E>>`? or better `From<Output> for Result<V, E>`?

Reply:
Ah, of course. Fixing instantly.